src/broadcast_receiver.cpp

```diff
@@ -32,14 +32,10 @@ void BroadcastReceiver::proceedReceivedData(const uint8_t* buffer,
     {
         case TESTER_PRESENT_REQ:
         {
-            if(buffer[1] == 0x80 && num_bytes >= 2)
-            {
-                pUdsReceiver_->pSessionCtrl_->reset();
-            }
-            else
+            if(buffer[1] != 0x80)
             {
+               pUdsReceiver_->pSessionCtrl_->reset();
```

Comment:
I guess this needs to go out of the 'if' block, because the timer should be reset no matter if Suppress-Pos-Response ist set or not.
Furthermore I'm not sure if the session timer needs to be reset only on TP or on any functional request.

Reply:
I'm pretty sure the S3 timer gets reset on reception of any diagnostic request.